macosx/PiecesView.mm

```diff
@@ -151,7 +151,7 @@ - (void)updateView
 
     // get the current state
     BOOL const showAvailability = [NSUserDefaults.standardUserDefaults boolForKey:@"PiecesViewShowAvailability"];
-    NSInteger const numCells = MIN(_torrent.pieceCount, kMaxCells);
+    int const numCells = static_cast<int>(MIN(_torrent.pieceCount, kMaxCells));
```

Comment:
What line was the warning generated on before?

Asking because I'm wondering if we can fix this warning in some other way.

I wouldn't block the PR on this change but it seems a little weird because we use `numCells` as an NSInteger on lines 166, 187, 191, 196.

Reply:
Simply the next two lines, 156 and 157. `tr_torrentAvailability` takes an int.

Storing an NSInteger in an int is a warning, so we need to cast (which is safe, since kMaxCells is 18x18).
Storing an int in an NSInteger is permitted, so no need to change lines 166 and later.

Reply:
Wouldn't this PR make the problem slightly worse then?

Torrent.h still declares the functions this way:

  - (void)getAvailability:(int8_t*)tab size:(NSInteger)size;
  - (void)getAmountFinished:(float*)tab size:(NSInteger)size;

... so it seems like we'd still  be getting warnings in Torrent.mm

```
- (void)getAvailability:(int8_t*)tab size:(NSInteger)size
{
    tr_torrentAvailability(self.fHandle, tab, size);
}

- (void)getAmountFinished:(float*)tab size:(NSInteger)size
{
    tr_torrentAmountFinished(self.fHandle, tab, size);
}
```

and then this PR would just add extra int <-> NSInteger conversions that didn't actually fix the warning?

Reply:
Check the changes again: Torrent.h declares it as `int` now, not `NSInteger`.

Reply:
Why not just change the call to `tr_torrentAvailability()` instead of mixing use of `int` and `NSInteger` in the macOS code?

e.g. 

```diff
(void)getAvailability:(int8_t*)tab size:(NSInteger)size
{
---    tr_torrentAvailability(self.fHandle, tab, size);
+++    tr_torrentAvailability(self.fHandle, tab, (int)size);
}
```

Reply:
No, this would be bad practice: one shall not cast to a smaller storage unless the code guaranties it's safe to do so.

If you just want to avoid seeing two types in use in the same function, I can simply change the remaining `NSInteger` to be `int`, even if unnecessary.

Reply:
(This won't be up for changing until the next semver-major, but the `int` in those two functions seems like a mistake in the first place -- probably `size_t` would've been better because neither of those should be accepting negative numbers.)

Reply:
> If you just want to avoid seeing two types in use in the same function, I can simply change the remaining NSInteger to be int, even if unnecessary.

No, I agree that's unnecessary :man_shrugging:

Reply:
Max value for this variable in macOS client is 18*18=324, so it fits in an int for now. So we can merge that PR safely and consider the `size_t` in the future.

Reply:
@ckerr good to merge?